buffer/buffer.mbti

```diff
@@ -16,10 +16,16 @@ impl T {
   write_byte(Self, Byte) -> Unit
   write_bytes(Self, Bytes) -> Unit
   write_char(Self, Char) -> Unit
+  write_double(Self, Double) -> Unit
+  write_float(Self, Float) -> Unit
+  write_int(Self, Int) -> Unit
+  write_int64(Self, Int64) -> Unit
   write_object(Self, Show) -> Unit
   write_string(Self, String) -> Unit
   write_sub_string(Self, String, Int, Int) -> Unit //deprecated
   write_substring(Self, String, Int, Int) -> Unit
+  write_uint(Self, UInt) -> Unit
+  write_uint64(Self, UInt64) -> Unit
 }
```

Comment:
we need mark the endianess, for example, in ocaml
```
val add_uint16_ne : t -> int -> unit
(** [add_uint16_ne b i] appends a binary native-endian unsigned 16-bit
    integer [i] to [b].
    @since 4.08
*)

val add_uint16_be : t -> int -> unit
(** [add_uint16_be b i] appends a binary big-endian unsigned 16-bit
    integer [i] to [b].
    @since 4.08
*)

val add_uint16_le : t -> int -> unit
(** [add_uint16_le b i] appends a binary little-endian unsigned 16-bit
    integer [i] to [b].
    @since 4.08
*)
```

Reply:
Added some endian variants methods. But for native endian, does MoonBit have a function to check it?

Reply:
I also notice that write_char and write_string does not provide different endian variants. Do I need to complete them?

Reply:
> I also notice that write_char and write_string does not provide different endian variants. Do I need to complete them?

Encoding-related APIs like `write_char` and `write_string` will be moved to x/encoding